dazeus-noms.py

```diff
@@ -139,57 +192,159 @@ def getMenu(dt=datetime.now()):
+    def isOpen(dt=datetime.now()):
+        # Monday to Friday: 08:00 until 20:00
+        if dt.weekday() < 5:
+            return dt.hour >= 8 and dt.hour < 20
+
+        # Weekends from 11:00 until 20:00
+        elif dt.weekday() >= 5:
+            return dt.hour >= 11 and dt.hour < 20
+
+        # ... What planet is this?
+        else:
+            raise ValueError('Encountered an unexpected weekday.')
 
     @staticmethod
-    def isOpenForLunch():
-        # Lunch wordt geserveerd van 11.30 tot 14.00 uur
-        pass
+    def willOpen(dt=datetime.now()):
+        if dt.weekday() < 5:
+            return dt.hour < 8
+        elif dt.weekday() >= 5:
+            return dt.hour < 11
+        else:
+            raise ValueError('Encountered an unexpected weekday.')
 
     @staticmethod
-    def isOpenForDinner():
-        # Diner van 16.30 tot 19.30 uur.
-        pass
+    def isOpenForLunch(dt=datetime.now()):
+        # Every day from 11.30 until 14.00
+        return (dt.hour == 11 and dt.minute >= 30) or (dt.hour > 11 and dt.hour < 14)
+
+    @staticmethod
+    def isOpenForDinner(dt=datetime.now()):
+        # Every day from 16.30 until 19.30.
+        (h, m) = (dt.hour, dt.minute)
+        return (h == 16 and m >= 30) or (h > 16 and h < 19) or (h == 19 and m < 30)
 
 
 class Refter(NomsSupplier):
+    friendly_name = 'De Refter'
+
     @staticmethod
     def getMenu(dt=datetime.now()):
         return []
 
     @staticmethod
-    def isOpen():
+    def isOpen(dt=datetime.now()):
         return False
 
     @staticmethod
-    def isOpenForLunch():
+    def isOpenForLunch(dt=datetime.now()):
         return Refter.isOpen()
 
     @staticmethod
-    def isOpenForDinner():
+    def isOpenForDinner(dt=datetime.now()):
         return Refter.isOpen()
 
 
 class DaZeusNoms(object):
+    suppliers = {
+        'gerecht': HetGerecht,
+        'umc': RadboudUMC,
+    }
+
     def __init__(self, address, verbose = False):
         self.dazeus = DaZeus(address, verbose)
         self.dazeus.subscribe_command("noms", self.handleMessage)
 
-    def handleMessage(self, event, message):
-        self.reply("Not actually implemented yet...")
-
-    def sendReply(self, msg):
-        self.reply(msg)
-
     def listen(self):
         self.dazeus.listen()
 
+    def handleMessage(self, event, reply):
+        # Ready available suppliers
+        suppliers = DaZeusNoms.suppliers.keys()
+        params = event['params']
+
+        # Ensure a restaurant is set.
+        if len(params) < 6:
+            reply('Where would you like to eat? I know the menu for: ' + ', '.join(suppliers))
+            return
+
+        # Ensure it is one we know.
+        supplier = params[5].lower()
+        if supplier not in suppliers:
+            reply('I don\'t have a menu for that place... How about one of these: ' + ', '.join(suppliers))
+            return
+
+        # Did we say when we'd like to eat?
+        now = datetime.now();
+        if len(params) >= 7:
```

Comment:
You already tested for len(params) < 6 above, so this will always be true here.

Reply:
This won't be true if len(params) == 6, when only a supplier name is provided, right? :)

Reply:
Oh, you're right, sorry!